patches/v8/0027-Remove-DCHECK-from-WriteOneByteV2-to-skip-v8-fatal.patch

```diff
@@ -0,0 +1,22 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Ketan Gupta <ketan@cloudflare.com>
+Date: Mon, 24 Mar 2025 12:29:19 +0000
+Subject: Remove DCHECK from WriteOneByteV2 to skip v8 fatal
+
+We already check for this after calling WriteOneByteV2()
+and throw a TypeError if the input string isn't one-byte only.
+
+diff --git a/src/api/api.cc b/src/api/api.cc
+index 43540a968a5b7e94c3e883d1bf8b5f51072bae05..da393be16944058da30acc9d0e3d94de0203257a 100644
+--- a/src/api/api.cc
++++ b/src/api/api.cc
+@@ -6157,7 +6157,8 @@ void String::WriteV2(Isolate* v8_isolate, uint32_t offset, uint32_t length,
+ 
+ void String::WriteOneByteV2(Isolate* v8_isolate, uint32_t offset,
+                             uint32_t length, uint8_t* buffer, int flags) const {
+-  DCHECK(IsOneByte());
```

Comment:
I think the assert should be ContainsOnlyOneByte instead, and I think we can uncomment it again if you do that. If not it seems like a bug in workerd that it is calling the API with non-Latin1 text.

The comment is wrong I think.

Probably a simple thing to upstream too.

Reply:
OK after off-line discussion, let's just commit this as is, then file bugs against both node and workerd to handle high bytes better.  Currently, node ignores the high bytes, while we incorrectly pass them to V8 and then redo all the work afterwards, which is wasteful but follows the spec (but not node).